### PR TITLE
Add MySQL REPLACE support

### PR DESCRIPTION
**Pull request by caulfield**
Hi.

This PR adds mysql REPLACE functionallity. 
[Docs](http://dev.mysql.com/doc/refman/5.7/en/replace.html) from mysql site.
I'm not very experienced with mysql and this gem, let me know if I missed something.

Up #54

### CODE REVIEW

**Comment by caulfield** [comment]
It's a very strange CI error. Do you have any thoughts? Everything is good locally

**Comment by jkowens** [comment]
@caulfield have you run into a situation were you needed REPLACE instead of UPSERT functionality?

**Comment by caulfield** [comment]
@jkowens, I created this pull request based on my current project requirements. If your database has a strict schema and you have no ability to change keys(primary). REPLACE helped me to keep consistency without running ruby code, it matters for performance and bulk inserts.
In my opinion, gem should provide an interface for communication with a database. Side effects(as an example of dangerous things) should be resolved by developers.
I understand that it's not a very popular feature, I can add more docs in README with links to MySQL docs.

**Comment by jkowens** [comment]
I still don't understand why ON DUPLICATE KEY UPDATE could not be used instead? Could you provide me with more detail about the reasons REPLACE would have to be used? 

Each feature that is added to a gem adds more code to support long term. If there isn't a strong reason to add a feature (or there is an existing alternative) then I think that is good reason to give pause and ask if it is really necessary.

I appreciate all the work that you've put into this patch, it is very thorough. If you can help me see a scenario where ON DUPLICATE KEY UPDATE could not be used in place of REPLACE, that may help sway my thinking on this addition.

**Comment by caulfield** [comment]
It makes sense with more code support.
The original problem was about inserting a lot of records without checking unique indices' values before. `REPLACE` gave me the more convenient way to insert new records. I receive a lot of time series data from 3rd party sources and I have to update(delete and then create new with default values) rows with duplicated keys. There weren't ids of records from this source, so I used a composite unique index.
I understand that `REPLACE` is less popular with `ON DUPLICATE KEY UPDATE` and the same task could be solved by both methods. The merge decision is up to you.

File: test/support/shared_examples/replace_import.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      CartItem.import columns, new_values, opts.extract_options!.merge(replace: true)
+    end
+
+    setup do
+      CartItem.import columns, values
+    end
+
+    it "should replace conflicting rows" do
+      perform_import
+      refute CartItem.find_by(id: 1).present?, "deletes conflicting rows"
```
Comment (jkowens) [inline test/support/shared_examples/replace_import.rb]: I think the use of `#find_by` is causing tests for AR 3.2 to fail.

**Comment by jkowens** [comment]
Closing for now.

**Pull request closed without merging**